scripts/pylib/twister/twisterlib.py

```diff
@@ -3830,11 +3830,13 @@ def json_report(self, filename, append=False, version="NA"):
                     if rom_size:
                         testcase["rom_size"] = rom_size
 
-                    if instance.results[k] in ["PASS"] or instance.status == 'passed':
+                    if instance.results[k] in ["SKIP"] or instance.status == 'skipped':
```

Comment:
Just personal suggestion: It would be good if we could add a comment here to descript why skip should be the first. Ex. The testcase which using ztest_test_skip(), its instance.status will be 'passed', that's why this bug happens.